go-controller/pkg/ovn/libovsdbops/nb_global.go

```diff
@@ -0,0 +1,62 @@
+package libovsdbops
+
+import (
+	"context"
+	"fmt"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+// findNBGlobal returns the NBGlobal table entry
+func findNBGlobal(nbClient libovsdbclient.Client) (*nbdb.NBGlobal, error) {
+	nbGlobal := []nbdb.NBGlobal{}
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+	err := nbClient.List(ctx, &nbGlobal)
+	if err != nil {
+		return nil, fmt.Errorf("failed listing nbGlobal table entires err: %v", err)
+	}
+	// We should error if the nbGlobal table entry does not exist
+	if len(nbGlobal) == 0 {
+		return nil, libovsdbclient.ErrNotFound
+	}
+	// The nbGlobal table should only have one row
+	if len(nbGlobal) != 1 {
+		return nil, fmt.Errorf("multible NBGlobal rows found")
+	}
+
+	return &nbGlobal[0], nil
+}
+
+// UpdateNBGlobalOptions updates the options on the NBGlobal table, adding any newly specified options in the process
+func UpdateNBGlobalOptions(nbClient libovsdbclient.Client, options map[string]string) error {
+	// find the nbGlobal table's UUID, we don't have any other way to reliably look this table entry since it can
+	// only be indexed by UUID
+	nbGlobal, err := findNBGlobal(nbClient)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range options {
+		nbGlobal.Options[k] = v
+	}
+
+	// Update the options column in the nbGlobal entry since we already performed a lookup
```

Comment:
ack. I think that is okay. a pathological case would be if another thread tried to do the same around the same time. But that is very unlikely, right? Mutates are nicer because they only mess with the values we care about at the transaction level at the server.

Reply:
Fair enough, in this case though there really shouldn't be many threads trying to update the go-routine, I could make it a mutate, It just adds an extra transaction where we don't necessarily need one.